misk/src/main/kotlin/misk/backoff/Retries.kt

```diff
@@ -0,0 +1,34 @@
+package misk.backoff
+
+/**
+ * Retries the provided function up to a certain number of times, applying the given backoff
+ * between each retry. The retry function is provided with current retry count, in case this is
+ * relevant
+ */
+fun <A> retry(upTo: Int, withBackoff: Backoff, f: (Int) -> A): A {
```

Comment:
it's not immediately obvious that the Int parameter to the function is the current retry number. any way to improve that?